src/rust/engine/fs/store/src/lib.rs

```diff
@@ -1394,31 +1395,51 @@ impl Store {
     perms: Permissions,
     is_executable: bool,
   ) -> Result<(), StoreError> {
-    self
-      .load_file_bytes_with(digest, move |bytes| {
-        let mut f = OpenOptions::new()
-          .create(true)
-          .write(true)
-          .truncate(true)
-          .mode(match perms {
-            Permissions::ReadOnly if is_executable => 0o555,
-            Permissions::ReadOnly => 0o444,
-            Permissions::Writable if is_executable => 0o755,
-            Permissions::Writable => 0o644,
-          })
-          .open(&destination)
+    let mode = match perms {
+      Permissions::ReadOnly if is_executable => 0o555,
+      Permissions::ReadOnly => 0o444,
+      Permissions::Writable if is_executable => 0o755,
+      Permissions::Writable => 0o644,
+    };
+    match self.local.load_from_fs(digest).await? {
+      Some(path) => {
+        tokio::fs::copy(path.clone(), destination.clone())
```

Comment:
These parameters are `AsRef<Path>`, meaning that they will accept anything which can be viewed as `&Path`. If the argument is _already_ a `&Path`, you don't have to give up ownership.
```suggestion
        tokio::fs::copy(&path, &destination)
```